diff --git a/node_modules/svelte-echarts/dist/Chart.svelte b/node_modules/svelte-echarts/dist/Chart.svelte
index 8495955..7785cec 100644
--- a/node_modules/svelte-echarts/dist/Chart.svelte
+++ b/node_modules/svelte-echarts/dist/Chart.svelte
@@ -9,7 +9,8 @@ export function chartable(element, echartOptions) {
     ...echartOptions
   };
   const echartsInstance = echarts.init(element, theme, { renderer });
-  echartsInstance.setOption(options);
+  const opts = { replaceMerge: ['xAxis', 'series'] };
+  echartsInstance.setOption(options, opts);
   function handleResize() {
     echartsInstance.resize();
   }
@@ -23,7 +24,7 @@ export function chartable(element, echartOptions) {
       echartsInstance.setOption({
         ...echartOptions.options,
         ...newOptions.options
-      });
+      }, opts);
     }
   };
 }
